### PR TITLE
Initial quick piece of work on NPS survey for 2015.

### PR DESCRIPTION
**Pull request by breville**

### CODE REVIEW

**Comment by laurelfan** [comment]
Looks like there are some test failures... I think because it doesn't work if there's no logged in user?

File: dashboard/app/views/home/index.html.haml [hunk 1]
```diff
@@ -1,3 +1,5 @@
+= render partial: 'layouts/survey'
```
Comment (laurelfan) [inline dashboard/app/views/home/index.html.haml]: I think changing this to `= render partial: 'layouts/survey' if current_user` will fix your test failures

File: dashboard/app/views/layouts/_survey.html.haml [hunk 2]
```diff
@@ -0,0 +1,67 @@
+-# If user is seeing English
+- if I18n.locale.to_s.downcase.split('-').first == "en"
```
Comment (laurelfan) [inline dashboard/app/views/layouts/_survey.html.haml]: We have a helper `language` that does this that you can use here: https://github.com/code-dot-org/code-dot-org/blob/5214eac47ab2eac67da55db29c10e9883d3381f5/dashboard/app/helpers/locale_helper.rb#L18

File: dashboard/app/views/layouts/_survey.html.haml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+  -# If user hasn't already filled out this survey
+  - if !current_user[:properties]['survey2015_value']
```
Comment (laurelfan) [inline dashboard/app/views/layouts/_survey.html.haml]: the serializedattributes stuff defines methods for you so you can use `current_user.survey2015_value`

File: dashboard/app/views/layouts/_survey.html.haml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+:javascript
+
+  function enableSurveySubmit()
+  {
+    $("#survey #textbox").slideDown();
+    $("#submitSurvey").prop("disabled",false);
+  }
+
+  window.submitSurvey = function(submittedValue)
```
Comment (laurelfan) [inline dashboard/app/views/layouts/_survey.html.haml]: why put one of these on window and the other one not?

File: dashboard/app/views/layouts/_survey.html.haml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        #surveybody
+          %h1{style: "margin-top:0px; color: rgb(80,80,80)"} Quick survey
+          %p{style: "padding-top: 10px; font-size:18px"}
+            How likely are you to recommend Code.org to a friend or colleague?
+          = form_for(current_user, url: registration_url('user'), html: {id: 'survey', onsubmit: "return window.submitSurvey(true)", style: "margin-bottom:0px"}) do |f|
+            #dots{style: "padding-top: 10px; padding-bottom: 20px"}
+              - (-1..10).each do |i|
+                - display = i == -1 ? "none" : "inline"
+                %input{style: "margin-right:3px; vertical-align:top; display: #{display}", id: "radio_#{i}", type: "radio", name: "user[survey2015_value]", value: i, onclick: "enableSurveySubmit()"}
+                %label{style: "display: #{display}; font-size: 20px; padding-right:20px", for: i}= i
```
Comment (laurelfan) [inline dashboard/app/views/layouts/_survey.html.haml]: for should be radio_#{i} to match the above, right?

File: dashboard/app/views/layouts/_survey.html.haml [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+      .survey{style: "background-color: rgb(200,200,200); padding:40px; font-size:18px"}
+        #surveythanks{style: "display: none"} Thank you for your response.
+        #surveybody
+          %h1{style: "margin-top:0px; color: rgb(80,80,80)"} Quick survey
+          %p{style: "padding-top: 10px; font-size:18px"}
+            How likely are you to recommend Code.org to a friend or colleague?
+          = form_for(current_user, url: registration_url('user'), html: {id: 'survey', onsubmit: "return window.submitSurvey(true)", style: "margin-bottom:0px"}) do |f|
+            #dots{style: "padding-top: 10px; padding-bottom: 20px"}
+              - (-1..10).each do |i|
```
Comment (laurelfan) [inline dashboard/app/views/layouts/_survey.html.haml]: random ruby trick, upto method: http://ruby-doc.org/core-2.2.0/Integer.html#method-i-upto

**Comment by breville** [comment]
Thanks @laurelfan ... made fixes based on your excellent feedback.  Also made it only show up 1/10 times.

**Pull request merged**